src/dbnode/generated/proto/namespace/namespace.proto

```diff
@@ -33,6 +33,31 @@ message NamespaceOptions {
     bool coldWritesEnabled                          = 10;
     NamespaceRuntimeOptions runtimeOptions          = 11;
     google.protobuf.BoolValue cacheBlocksOnRetrieve = 12;
+    AggregationOptions aggregationOptions           = 13;
+}
+
+message AggregationOptions {
+    repeated Aggregation aggregations = 1;
+}
+
+message Aggregation {
+    AggregationAttributes attributes    = 1;
+    DownsampleOptions downsampleOptions = 2;
+}
+
+message AggregationAttributes {
+    int64 resolutionNanos = 1;
```

Comment:
what does this field mean when UNAGGREGATED is set?

Reply:
It's unused. More details are in the comments on the Golang types: https://github.com/m3db/m3/pull/2661/files#diff-3b923c7bae83be97b81cd22f2cfe3584R340

Reply:
This may not be worth it, but one option to restrict the relevant fields by type would be to make the `attributes` use `oneof` proto feature and remove the `aggregationType`:
```
message Aggregation {
	oneof attributes {
            UnaggregatedAttributes unaggregated_attributes = 1;
            AggregatedAttributes aggregated_attributes = 2;
        }
	DownsampleOptions downsampleOptions  = 3;
}

message AggregatedAttributes {
	int64 resolutionNanos   = 1;
}

message UnaggregatedAttributes {}
```

Reply:
I think I'm gonna move into the other direction and simplify this. Going to take @ryanhall07 's suggestion and flatten out aggregated attributes to be directly on the aggregation and convert the enum to a boolean. I think we can get away with commenting that resolution is ignored in the unaggregated case.

Reply:
👍